src/proxy/inbound.rs

```diff
@@ -67,7 +67,11 @@ impl Inbound {
             .map_err(|e| Error::Bind(pi.cfg.inbound_addr, e))?;
         let transparent = super::maybe_set_transparent(&pi, &listener)?;
         // Override with our explicitly configured setting
-        pi.cfg.enable_original_source = Some(transparent);
+        if pi.cfg.enable_original_source.is_none() {
+            let mut cfg = (*pi.cfg).clone();
```

Comment:
Rather than create a mutable clone of the whole config we might as well just hoist `let enable_original_source = pi.cfg.enable_original_source;`?

We don't need to put the value back in the config, AFAIK.

Reply:
yeah we use it somewhat all over though.. I was originally thinking that then went with this approach. But I hate it as well, I can change it up

Reply:
This still feels suboptimal but it would probably take more refactoring to fix, and it's an improvement in some cases at least.